service/internal/endpoint-discovery/go.mod

```diff
@@ -0,0 +1,12 @@
+module github.com/aws/aws-sdk-go-v2/service/internal/endpoint-discovery
+
+go 1.15
+
+require (
+	github.com/aws/aws-sdk-go-v2/config v1.1.7
+	github.com/aws/smithy-go v1.4.0
+)
+
+replace (
+	github.com/aws/aws-sdk-go-v2/config => ./../../../config
+)
```

Comment:
This runs the risk of creating circular dependencies.  For example if STS were to ever add endpoint discovery this would break.

`config` -> `credentials/stscreds` -> `service/sts` -> `service/internal/endpoint-discovery` -> `config`

Nothing in the SDK may depend on `config` module.